llvm/test/CodeGen/AMDGPU/llvm.amdgcn.implicit.ptr.buffer.ll

```diff
@@ -16,8 +16,8 @@ define amdgpu_ps i32 @test_ps() #1 {
 }
 
 ; GCN-LABEL: {{^}}test_cs:
-; GCN: s_mov_b64 s[4:5], s[0:1]
-; GCN: buffer_store_dword v{{[0-9]+}}, off, s[4:7], 0 offset:4
+; GCN: s_mov_b64 s[8:9], s[0:1]
+; GCN: buffer_store_dword v{{[0-9]+}}, off, s[8:11], 0 offset:4
```

Comment:
Is this expected?

Reply:
No, I guess I've broken something.

Reply:
The following change? It now enables workgroup IDs for AMDGPU_CS always.
SIMachineFunctionInfo.cpp:110
  - if (!AMDGPU::isGraphics(CC) ||
  -    (CC == CallingConv::AMDGPU_CS && ST.hasArchitectedSGPRs())) {
  
  +  if (!AMDGPU::isGraphics(CC) || CC == CallingConv::AMDGPU_CS ||
  +    ST.hasArchitectedSGPRs()) {